### PR TITLE
Move filtergraph functionality to new module

### PR DESCRIPTION
**Pull request by Erik-White**
Refactor and move filtering code to a separate module to improve maintainability and allow easier reuse when implementing new features.

### CODE REVIEW

File: src/libvx.c [hunk 1]
```diff
@@ -20,33 +16,21 @@
...
+#include "util.h"
+#include "filter.h"
+#include "filtergraph.h"
 
 #ifdef __cplusplus
 #pragma error
 #endif
 
-#ifdef DEBUG
-#define dprintf(...) { printf("%s:%d %-30s ", __FILE__, __LINE__, __func__); printf(__VA_ARGS__); }
```
Comment (Erik-White) [inline src/libvx.c:29]: Replaced with `av_log`

File: src/libvx.c [hunk 2]
```diff
@@ -1717,22 +1218,15 @@ vx_error vx_frame_transfer_data(const vx_video* video, vx_frame* frame)
...
+	if ((ret = vx_frame_properties_from_metadata(frame, av_frame)) != VX_ERR_SUCCESS) {
 		goto cleanup;
 	}
 
-	return VX_ERR_SUCCESS;
-
 cleanup:
-	if (av_frame) {
-		av_frame_unref(av_frame);
-		av_frame_free(&av_frame);
```
Comment (Erik-White) [inline src/libvx.c:1733]: If an error occurs and this point is reached then the frame should not be cleared here, it is cleaned up when the video is closed.

File: src/filtergraph.c [hunk 3]
```diff
@@ -0,0 +1,138 @@
+#include <stdlib.h>
+#include <stdbool.h>
+
+#include <libavcodec/avcodec.h>
+#include <libavfilter/avfilter.h>
+#include <libavfilter/buffersink.h>
+#include <libavfilter/buffersrc.h>
+#include <libavformat/avformat.h>
+
+#include "libvx.h"
```
Comment (klerebladh) [inline src/filtergraph.c:10]: Long time size I coded c/c++ :) But I was wondering about these includes... if they should be in the header file instead size the function definitions there use types defined in some of these h-files?

**Review by klerebladh — COMMENTED** [review]
Nice improvement!

File: src/filtergraph.c (hunk 3 above)
Comment (Erik-White) [reply src/filtergraph.c:10]: There are different conventions, and a lot of different viewpoints on this topic! I will leave it as it is for now, but follow up after I have done some more research.

**Pull request merged**